galaxy_importer/collection.py

```diff
@@ -49,10 +50,17 @@ def import_collection(file, filename=None, logger=None):
 
 
 def _import_collection(file, filename, logger):
-    with tempfile.TemporaryDirectory() as extract_dir:
+    with tarfile.open(fileobj=file, mode='r') as pkg_tar:
+        manifest_file = pkg_tar.extractfile('MANIFEST.json')
```

Comment:
Check for an exception here in case `MANIFEST.json` does not exist or cannot be extracted for some reason.

Reply:
Resolved by checking `MANIFEST.json` only once which includes those validation checks